packages/components/file-uploader/src/file-uploader.css

```diff
@@ -0,0 +1,421 @@
+  font-weight: 600;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+}
+
+:host > .file-card-wrapper > .file-card--failed .file-upload--info,
+:host > .file-card-wrapper > .file-card--download > .file-upload--info > div {
+  line-height: 14px;
+}
+
+:host > .file-card-wrapper > .file-card--failed small,
+:host > .file-card-wrapper > .file-card--download small {  
+  font-size: calc(1px + var(--ts-unit-half));
+}
+
+:host > .file-card-wrapper > .file-card--failed small:first-of-type,
+:host > .file-card-wrapper > .file-card--download small {
+  color: #7F7F7F;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+}
+
+/* Uploading state................................................... */
+
+:host > .file-card-wrapper > .file-card {
+  cursor: default;
+  height: calc(6.5 * var(--ts-unit-half));
+  margin-bottom: 0;
+}
+
+:host > .file-card-wrapper > .file-card:hover {
+  border: var(--ts-border);
+}
+
+:host > .file-card-wrapper .icon { 
+  align-items: center;
+  border-radius: 50%;
+  color: var(--ts-color-white);
+  display: flex;
+  justify-content: center;
+}
+
+:host > .file-card-wrapper > .file-card .icon:first-of-type {
+  background-color: var(--ts-color-blue);
+  height: 28px;
+  width: 28px;
+}
+
+:host > .file-card-wrapper .icon:first-of-type > span {
+  font-size: var(--ts-unit);
+  font-weight: bold;
+}
+
+:host > .file-card-wrapper .file-upload--info,
+:host > .file-card--download .file-upload--info {
+  display: flex;
+  flex-direction: column;
+  max-width: calc(16.5 * var(--ts-unit-half));
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+}
+
+:host > .file-card-wrapper > .file-card--uploading .progress {
+  height: calc(0.5 * var(--ts-unit-half));
+  margin-top: calc(0.5 * var(--ts-unit-half));
+  position: relative;
+  width: calc(16.5 * var(--ts-unit-half));
+}
+
+:host > .file-card-wrapper > .file-card--uploading .progress > div {
+  border-radius: calc(0.5 * var(--ts-unit-half));
+  height: 100%;
+  position: absolute;
+  width: 100%;
+}
+
+:host > .file-card-wrapper > .file-card--uploading .progress > .track {
+  background-color: var(--ts-color-blue-lighter);
+}
+
+:host > .file-card-wrapper > .file-card--uploading .progress > .indicator {
+  animation-name: progress;
+  animation-duration: 1.25s;
+  animation-iteration-count: infinite;
+  animation-timing-function: linear;
+  background-color: var(--ts-color-blue);
+  width: 50px;
+  z-index: 2;
+}
+
+@keyframes progress {
+  0% {
+    left: 0;
+    width: calc(2.5 * var(--ts-unit-half));
+  }
+  25% {
+    left: 0;
+    width: calc(8 * var(--ts-unit-half));
+  }
+  50% {
+    left: calc(5 * var(--ts-unit-half));
+    width: calc(12 * var(--ts-unit-half));
+  }
+  75% {
+    left: calc(13 * var(--ts-unit-half));
+    width: calc(15 * var(--ts-unit-half));
+  }
+  100% {
+    left: 150%;
+    width: 0;
+  }
+}
+
+:host > .file-card-wrapper > .file-card--uploading label {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Simplify this over-specified selector. Maximum allowed depth: 3. Actual depth: 4 [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=css%3Aoverspecific-selectors)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this usage of the experimental "host" pseudo. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=css%3Aexperimental-pseudo-usage)